sources/launcher/Stride.Launcher/Services/SelfUpdater.cs

```diff
@@ -214,10 +214,13 @@ internal static async Task DownloadAndInstallNewVersion(IDispatcherService dispa
 
 
                 var strideInstaller = Path.Combine(Path.GetTempPath(), $"StrideSetup-{Guid.NewGuid()}.exe");
-                using (WebClient webClient = new WebClient())
-                {
-                    webClient.DownloadFile(strideInstallerUrl, strideInstaller);
-                }
+                var response = await httpClient.GetAsync(strideInstallerUrl);
+                response.EnsureSuccessStatusCode();
+
+                await using var responseStream = await response.Content.ReadAsStreamAsync();
+                await using var fileStream = File.Create(strideInstaller);
+                responseStream.Seek(0, SeekOrigin.Begin);
```

Comment:
Do you need to seek here? Since no other reads happened on the `responseStream` this can be avoided (and prevent an issue if in the future the stream isn't seekable).

Reply:
No, we don't.  I had a moment of weakness and left it there. :)   Thanks for the clarity.

Reply:
Removed in second commit.